ServerGroup Legacy Microversion not working

### Affected Resource(s)
- openstack_compute_servergroup_v2

### Terraform Configuration Files
```hcl
resource "openstack_compute_servergroup_v2" "this" {
  name     = "this"
  policies = ["soft-anti-affinity"]
}
```

### Expected Behavior
ServerGroup with Name "this" and policy "soft-anti-affinity" should be created.

### Actual Behavior
Terraform can't create the ServerGroup: `{"badRequest": {"message": "Invalid input for field/attribute 0. Value: soft-anti-affinity. u'soft-anti-affinity' is not one of ['anti-affinity', 'affinity']", "code": 400}}`

### Important Factoids
OpenStack microversion maximum is 2.60. This should be no problem, due to legacy check (see linked PR below). But i guess the legacy check is not working. 
With provider version 1.50 (before release of the PR down below) everything works as expected. 

### References
- https://github.com/terraform-provider-openstack/terraform-provider-openstack/pull/1382

Thanks for your reply.

You can find my debug log (with removed login info and urls) here:
```
2023-03-20T11:48:58.752+0100 [INFO]  backend/local: apply calling Apply
2023-03-20T11:48:58.752+0100 [DEBUG] Building and walking apply graph for NormalMode plan
2023-03-20T11:48:58.752+0100 [DEBUG] Resource state not found for node "openstack_compute_servergroup_v2.this", instance openstack_compute_servergroup_v2.this
2023-03-20T11:48:58.752+0100 [DEBUG] ProviderTransformer: "openstack_compute_servergroup_v2.this (expand)" (*terraform.nodeExpandApplyableResource) needs provider["registry.terraform.io/terraform-provider-openstack/openstack"]
2023-03-20T11:48:58.752+0100 [DEBUG] ProviderTransformer: "openstack_compute_servergroup_v2.this" (*terraform.NodeApplyableResourceInstance) needs provider["registry.terraform.io/terraform-provider-openstack/openstack"]
2023-03-20T11:48:58.752+0100 [DEBUG] ReferenceTransformer: "provider[\"registry.terraform.io/terraform-provider-openstack/openstack\"]" references: []
2023-03-20T11:48:58.752+0100 [DEBUG] ReferenceTransformer: "openstack_compute_servergroup_v2.this (expand)" references: []
2023-03-20T11:48:58.752+0100 [DEBUG] ReferenceTransformer: "openstack_compute_servergroup_v2.this" references: []
2023-03-20T11:48:58.752+0100 [DEBUG] Starting graph walk: walkApply
2023-03-20T11:48:58.753+0100 [DEBUG] created provider logger: level=debug
2023-03-20T11:48:58.753+0100 [INFO]  provider: configuring client automatic mTLS
2023-03-20T11:48:58.762+0100 [DEBUG] provider: starting plugin: path=.terraform/providers/registry.terraform.io/terraform-provider-openstack/openstack/1.51.0/linux_amd64/terraform-provider-openstack_v1.51.0 args=[.terraform/providers/registry.terraform.io/terraform-provider-openstack/openstack/1.51.0/linux_amd64/terraform-provider-openstack_v1.51.0]
2023-03-20T11:48:58.763+0100 [DEBUG] provider: plugin started: path=.terraform/providers/registry.terraform.io/terraform-provider-openstack/openstack/1.51.0/linux_amd64/terraform-provider-openstack_v1.51.0 pid=23317
2023-03-20T11:48:58.764+0100 [DEBUG] provider: waiting for RPC address: path=.terraform/providers/registry.terraform.io/terraform-provider-openstack/openstack/1.51.0/linux_amd64/terraform-provider-openstack_v1.51.0
2023-03-20T11:48:58.772+0100 [INFO]  provider.terraform-provider-openstack_v1.51.0: configuring server automatic mTLS: timestamp=2023-03-20T11:48:58.772+0100
2023-03-20T11:48:58.813+0100 [DEBUG] provider.terraform-provider-openstack_v1.51.0: plugin address: address=/tmp/plugin4270407996 network=unix timestamp=2023-03-20T11:48:58.813+0100
2023-03-20T11:48:58.813+0100 [DEBUG] provider: using plugin: version=5
2023-03-20T11:48:58.877+0100 [WARN]  ValidateProviderConfig from "provider[\"registry.terraform.io/terraform-provider-openstack/openstack\"]" changed the config value, but that value is unused
2023-03-20T11:48:58.879+0100 [INFO]  provider.terraform-provider-openstack_v1.51.0: 2023/03/20 11:48:58 [DEBUG] OpenStack allowReauth: true: timestamp=2023-03-20T11:48:58.879+0100
2023-03-20T11:48:58.882+0100 [WARN]  Provider "registry.terraform.io/terraform-provider-openstack/openstack" produced an invalid plan for openstack_compute_servergroup_v2.this, but we are tolerating it because it is using the legacy plugin SDK.
    The following problems may be the cause of any confusing errors from downstream operations:
      - .rules: attribute representing nested block must not be unknown itself; set nested attribute values to unknown instead
openstack_compute_servergroup_v2.this: Creating...
2023-03-20T11:48:58.882+0100 [INFO]  Starting apply for openstack_compute_servergroup_v2.this
2023-03-20T11:48:58.883+0100 [DEBUG] openstack_compute_servergroup_v2.this: applying the planned Create change
2023-03-20T11:48:58.886+0100 [INFO]  provider.terraform-provider-openstack_v1.51.0: 2023/03/20 11:48:58 [DEBUG] setting computed for "rules" from ComputedKeys: timestamp=2023-03-20T11:48:58.884+0100
2023-03-20T11:48:58.886+0100 [INFO]  provider.terraform-provider-openstack_v1.51.0: 2023/03/20 11:48:58 [DEBUG] setting computed for "members" from ComputedKeys: timestamp=2023-03-20T11:48:58.884+0100
2023-03-20T11:48:58.886+0100 [INFO]  provider.terraform-provider-openstack_v1.51.0: 2023/03/20 11:48:58 [DEBUG] Locking "auth": timestamp=2023-03-20T11:48:58.884+0100
2023-03-20T11:48:58.886+0100 [INFO]  provider.terraform-provider-openstack_v1.51.0: 2023/03/20 11:48:58 [DEBUG] Locked "auth": timestamp=2023-03-20T11:48:58.884+0100
2023-03-20T11:48:58.886+0100 [INFO]  provider.terraform-provider-openstack_v1.51.0: 2023/03/20 11:48:58 [DEBUG] OpenStack Request URL: POST https://MY_OPENSTACK_API.com/v3/auth/tokens: timestamp=2023-03-20T11:48:58.884+0100
2023-03-20T11:48:58.886+0100 [INFO]  provider.terraform-provider-openstack_v1.51.0: 2023/03/20 11:48:58 [DEBUG] OpenStack Request Headers:
Accept: application/json
Cache-Control: no-cache
Content-Type: application/json
User-Agent: HashiCorp Terraform/1.3.7 (+https://www.terraform.io) Terraform Plugin SDK/2.10.1 gophercloud/v1.2.0: timestamp=2023-03-20T11:48:58.884+0100
2023-03-20T11:48:58.886+0100 [INFO]  provider.terraform-provider-openstack_v1.51.0: 2023/03/20 11:48:58 [DEBUG] OpenStack Request Body: {
  OPENSTACK_LOGIN_INFO_REMOVED
}: timestamp=2023-03-20T11:48:58.884+0100
2023-03-20T11:48:59.618+0100 [INFO]  provider.terraform-provider-openstack_v1.51.0: 2023/03/20 11:48:59 [DEBUG] OpenStack Response Code: 201: timestamp=2023-03-20T11:48:59.618+0100
2023-03-20T11:48:59.618+0100 [INFO]  provider.terraform-provider-openstack_v1.51.0: 2023/03/20 11:48:59 [DEBUG] OpenStack Response Headers:
Content-Length: 11131
Content-Type: application/json
Date: Mon, 20 Mar 2023 10:48:58 GMT
Server: Apache
Vary: X-Auth-Token
X-Openstack-Request-Id: req-43c2bf39-2167-407d-b5bd-0ff17d90aadd
X-Subject-Token: ***: timestamp=2023-03-20T11:48:59.618+0100
2023-03-20T11:48:59.618+0100 [INFO]  provider.terraform-provider-openstack_v1.51.0: 2023/03/20 11:48:59 [DEBUG] OpenStack Response Body: {
  OPENSTACK_LOGIN_INFO_REMOVED
}: timestamp=2023-03-20T11:48:59.618+0100
2023-03-20T11:48:59.619+0100 [INFO]  provider.terraform-provider-openstack_v1.51.0: 2023/03/20 11:48:59 [DEBUG] Unlocking "auth": timestamp=2023-03-20T11:48:59.619+0100
2023-03-20T11:48:59.619+0100 [INFO]  provider.terraform-provider-openstack_v1.51.0: 2023/03/20 11:48:59 [DEBUG] Unlocked "auth": timestamp=2023-03-20T11:48:59.619+0100
2023-03-20T11:48:59.619+0100 [INFO]  provider.terraform-provider-openstack_v1.51.0: 2023/03/20 11:48:59 [DEBUG] OpenStack Region is: regionOne: timestamp=2023-03-20T11:48:59.619+0100
2023-03-20T11:48:59.619+0100 [INFO]  provider.terraform-provider-openstack_v1.51.0: 2023/03/20 11:48:59 [DEBUG] OpenStack Endpoint for compute: https://MY_OPENSTACK_API.com/v2.1/: timestamp=2023-03-20T11:48:59.619+0100
2023-03-20T11:48:59.619+0100 [INFO]  provider.terraform-provider-openstack_v1.51.0: 2023/03/20 11:48:59 [DEBUG] openstack_compute_servergroup_v2 create options: openstack.ComputeServerGroupV2CreateOpts{CreateOpts:servergroups.CreateOpts{Name:"this", Policies:[]string(nil), Policy:"soft-anti-affinity", Rules:(*servergroups.Rules)(nil)}, ValueSpecs:map[string]string{}}: timestamp=2023-03-20T11:48:59.619+0100
2023-03-20T11:48:59.619+0100 [INFO]  provider.terraform-provider-openstack_v1.51.0: 2023/03/20 11:48:59 [DEBUG] OpenStack Request URL: POST https://MY_OPENSTACK_API.com/v2.1/os-server-groups: timestamp=2023-03-20T11:48:59.619+0100
2023-03-20T11:48:59.620+0100 [INFO]  provider.terraform-provider-openstack_v1.51.0: 2023/03/20 11:48:59 [DEBUG] OpenStack Request Headers:
Accept: application/json
Cache-Control: no-cache
Content-Type: application/json
Openstack-Api-Version: compute 2.64
User-Agent: HashiCorp Terraform/1.3.7 (+https://www.terraform.io) Terraform Plugin SDK/2.10.1 gophercloud/v1.2.0
X-Auth-Token: ***
X-Openstack-Nova-Api-Version: 2.64: timestamp=2023-03-20T11:48:59.619+0100
2023-03-20T11:48:59.620+0100 [INFO]  provider.terraform-provider-openstack_v1.51.0: 2023/03/20 11:48:59 [DEBUG] OpenStack Request Body: {
  "server_group": {
    "name": "this",
    "policy": "soft-anti-affinity"
  }
}: timestamp=2023-03-20T11:48:59.619+0100
2023-03-20T11:49:00.631+0100 [INFO]  provider.terraform-provider-openstack_v1.51.0: 2023/03/20 11:49:00 [DEBUG] OpenStack Response Code: 406: timestamp=2023-03-20T11:49:00.631+0100
2023-03-20T11:49:00.632+0100 [INFO]  provider.terraform-provider-openstack_v1.51.0: 2023/03/20 11:49:00 [DEBUG] OpenStack Response Headers:
Content-Length: 123
Content-Type: application/json; charset=UTF-8
Date: Mon, 20 Mar 2023 10:48:59 GMT
Openstack-Api-Version: compute 2.64
Server: Apache
Vary: OpenStack-API-Version,X-OpenStack-Nova-API-Version
X-Compute-Request-Id: req-61e1b291-405f-4abf-b0db-0c0b84771d3c
X-Openstack-Nova-Api-Version: 2.64
X-Openstack-Request-Id: req-61e1b291-405f-4abf-b0db-0c0b84771d3c: timestamp=2023-03-20T11:49:00.631+0100
2023-03-20T11:49:00.632+0100 [INFO]  provider.terraform-provider-openstack_v1.51.0: 2023/03/20 11:49:00 [DEBUG] OpenStack Response Body: {
  "computeFault": {
    "code": 406,
    "message": "Version 2.64 is not supported by the API. Minimum is 2.1 and maximum is 2.60."
  }
}: timestamp=2023-03-20T11:49:00.631+0100
2023-03-20T11:49:00.632+0100 [INFO]  provider.terraform-provider-openstack_v1.51.0: 2023/03/20 11:49:00 [DEBUG] Falling back to legacy API call due to: gophercloud.ErrUnexpectedResponseCode{BaseError:gophercloud.BaseError{DefaultErrString:"", Info:""}, URL:"https://MY_OPENSTACK_API.com/v2.1/os-server-groups", Method:"POST", Expected:[]int{200}, Actual:406, Body:[]uint8{0x7b, 0x22, 0x63, 0x6f, 0x6d, 0x70, 0x75, 0x74, 0x65, 0x46, 0x61, 0x75, 0x6c, 0x74, 0x22, 0x3a, 0x20, 0x7b, 0x22, 0x6d, 0x65, 0x73, 0x73, 0x61, 0x67, 0x65, 0x22, 0x3a, 0x20, 0x22, 0x56, 0x65, 0x72, 0x73, 0x69, 0x6f, 0x6e, 0x20, 0x32, 0x2e, 0x36, 0x34, 0x20, 0x69, 0x73, 0x20, 0x6e, 0x6f, 0x74, 0x20, 0x73, 0x75, 0x70, 0x70, 0x6f, 0x72, 0x74, 0x65, 0x64, 0x20, 0x62, 0x79, 0x20, 0x74, 0x68, 0x65, 0x20, 0x41, 0x50, 0x49, 0x2e, 0x20, 0x4d, 0x69, 0x6e, 0x69, 0x6d, 0x75, 0x6d, 0x20, 0x69, 0x73, 0x20, 0x32, 0x2e, 0x31, 0x20, 0x61, 0x6e, 0x64, 0x20, 0x6d, 0x61, 0x78, 0x69, 0x6d, 0x75, 0x6d, 0x20, 0x69, 0x73, 0x20, 0x32, 0x2e, 0x36, 0x30, 0x2e, 0x22, 0x2c, 0x20, 0x22, 0x63, 0x6f, 0x64, 0x65, 0x22, 0x3a, 0x20, 0x34, 0x30, 0x36, 0x7d, 0x7d}, ResponseHeader:http.Header{"Content-Length":[]string{"123"}, "Content-Type":[]string{"application/json; charset=UTF-8"}, "Date":[]string{"Mon, 20 Mar 2023 10:48:59 GMT"}, "Openstack-Api-Version":[]string{"compute 2.64"}, "Server":[]string{"Apache"}, "Vary":[]string{"OpenStack-API-Version,X-OpenStack-Nova-API-Version"}, "X-Compute-Request-Id":[]string{"req-61e1b291-405f-4abf-b0db-0c0b84771d3c"}, "X-Openstack-Nova-Api-Version":[]string{"2.64"}, "X-Openstack-Request-Id":[]string{"req-61e1b291-405f-4abf-b0db-0c0b84771d3c"}}}: timestamp=2023-03-20T11:49:00.632+0100
2023-03-20T11:49:00.632+0100 [INFO]  provider.terraform-provider-openstack_v1.51.0: 2023/03/20 11:49:00 [DEBUG] openstack_compute_servergroup_v2 create options: openstack.ComputeServerGroupV2CreateOpts{CreateOpts:servergroups.CreateOpts{Name:"this", Policies:[]string{"soft-anti-affinity"}, Policy:"", Rules:(*servergroups.Rules)(nil)}, ValueSpecs:map[string]string{}}: timestamp=2023-03-20T11:49:00.632+0100
2023-03-20T11:49:00.632+0100 [INFO]  provider.terraform-provider-openstack_v1.51.0: 2023/03/20 11:49:00 [DEBUG] OpenStack Request URL: POST https://MY_OPENSTACK_API.com/v2.1/os-server-groups: timestamp=2023-03-20T11:49:00.632+0100
2023-03-20T11:49:00.632+0100 [INFO]  provider.terraform-provider-openstack_v1.51.0: 2023/03/20 11:49:00 [DEBUG] OpenStack Request Headers:
Accept: application/json
Cache-Control: no-cache
Content-Type: application/json
User-Agent: HashiCorp Terraform/1.3.7 (+https://www.terraform.io) Terraform Plugin SDK/2.10.1 gophercloud/v1.2.0
X-Auth-Token: ***: timestamp=2023-03-20T11:49:00.632+0100
2023-03-20T11:49:00.632+0100 [INFO]  provider.terraform-provider-openstack_v1.51.0: 2023/03/20 11:49:00 [DEBUG] OpenStack Request Body: {
  "server_group": {
    "name": "this",
    "policies": [
      "soft-anti-affinity"
    ]
  }
}: timestamp=2023-03-20T11:49:00.632+0100
2023-03-20T11:49:00.667+0100 [INFO]  provider.terraform-provider-openstack_v1.51.0: 2023/03/20 11:49:00 [DEBUG] OpenStack Response Code: 400: timestamp=2023-03-20T11:49:00.667+0100
2023-03-20T11:49:00.667+0100 [INFO]  provider.terraform-provider-openstack_v1.51.0: 2023/03/20 11:49:00 [DEBUG] OpenStack Response Headers:
Content-Length: 173
Content-Type: application/json; charset=UTF-8
Date: Mon, 20 Mar 2023 10:49:00 GMT
Openstack-Api-Version: compute 2.1
Server: Apache
Vary: OpenStack-API-Version,X-OpenStack-Nova-API-Version
X-Compute-Request-Id: req-b2016e59-c5cc-430e-8fa0-a6556d53e3a7
X-Openstack-Nova-Api-Version: 2.1
X-Openstack-Request-Id: req-b2016e59-c5cc-430e-8fa0-a6556d53e3a7: timestamp=2023-03-20T11:49:00.667+0100
2023-03-20T11:49:00.667+0100 [INFO]  provider.terraform-provider-openstack_v1.51.0: 2023/03/20 11:49:00 [DEBUG] OpenStack Response Body: {
  "badRequest": {
    "code": 400,
    "message": "Invalid input for field/attribute 0. Value: soft-anti-affinity. u'soft-anti-affinity' is not one of ['anti-affinity', 'affinity']"
  }
}: timestamp=2023-03-20T11:49:00.667+0100
2023-03-20T11:49:00.667+0100 [ERROR] provider.terraform-provider-openstack_v1.51.0: Response contains error diagnostic: tf_req_id=08b132b5-1431-378e-1a40-90772c123a04 @caller=github.com/hashicorp/terraform-plugin-go@v0.14.3/tfprotov5/internal/diag/diagnostics.go:55 diagnostic_detail= diagnostic_summary="Error creating openstack_compute_servergroup_v2 this: Bad request with: [POST https://MY_OPENSTACK_API.com/v2.1/os-server-groups], error message: {"badRequest": {"message": "Invalid input for field/attribute 0. Value: soft-anti-affinity. u'soft-anti-affinity' is not one of ['anti-affinity', 'affinity']", "code": 400}}" tf_proto_version=5.3 tf_provider_addr=registry.terraform.io/terraform-provider-openstack/openstack @module=sdk.proto diagnostic_severity=ERROR tf_resource_type=openstack_compute_servergroup_v2 tf_rpc=ApplyResourceChange timestamp=2023-03-20T11:49:00.667+0100
2023-03-20T11:49:00.674+0100 [ERROR] vertex "openstack_compute_servergroup_v2.this" error: Error creating openstack_compute_servergroup_v2 this: Bad request with: [POST https://MY_OPENSTACK_API.com/v2.1/os-server-groups], error message: {"badRequest": {"message": "Invalid input for field/attribute 0. Value: soft-anti-affinity. u'soft-anti-affinity' is not one of ['anti-affinity', 'affinity']", "code": 400}}
╷
│ Error: Error creating openstack_compute_servergroup_v2 this: Bad request with: [POST https://MY_OPENSTACK_API.com/v2.1/os-server-groups], error message: {"badRequest": {"message": "Invalid input for field/attribute 0. Value: soft-anti-affinity. u'soft-anti-affinity' is not one of ['anti-affinity', 'affinity']", "code": 400}}
│
│   with openstack_compute_servergroup_v2.this,
│   on test.tf line 14, in resource "openstack_compute_servergroup_v2" "this":
│   14: resource "openstack_compute_servergroup_v2" "this" {
│
╵
2023-03-20T11:49:00.680+0100 [DEBUG] provider.stdio: received EOF, stopping recv loop: err="rpc error: code = Unavailable desc = error reading from server: EOF"
2023-03-20T11:49:00.682+0100 [DEBUG] provider: plugin process exited: path=.terraform/providers/registry.terraform.io/terraform-provider-openstack/openstack/1.51.0/linux_amd64/terraform-provider-openstack_v1.51.0 pid=23317
2023-03-20T11:49:00.682+0100 [DEBUG] provider: plugin exited
```

--- a/openstack/resource_openstack_compute_servergroup_v2.go
+++ b/openstack/resource_openstack_compute_servergroup_v2.go
@@ -163,16 +163,6 @@ func resourceComputeServerGroupV2Read(_ context.Context, d *schema.ResourceData,
 		if err != nil {
 			return diag.FromErr(CheckDeleted(d, err, "Error retrieving openstack_compute_servergroup_v2"))
 		}
-
-		log.Printf("[DEBUG] Retrieved openstack_compute_servergroup_v2 %s: %#v", d.Id(), sg)
-
-		d.Set("name", sg.Name)
-		d.Set("members", sg.Members)
-		d.Set("region", GetRegion(d, config))
-		d.Set("policies", sg.Policies)
-		d.Set("rules", nil)
-
-		return nil
 	}
 
 	log.Printf("[DEBUG] Retrieved openstack_compute_servergroup_v2 %s: %#v", d.Id(), sg)
@@ -180,10 +170,10 @@ func resourceComputeServerGroupV2Read(_ context.Context, d *schema.ResourceData,
 	d.Set("name", sg.Name)
 	d.Set("members", sg.Members)
 	d.Set("region", GetRegion(d, config))
-	if sg.Policy != nil {
+	if sg.Policy != nil && *sg.Policy != "" {
 		d.Set("policies", []string{*sg.Policy})
 	} else {
-		d.Set("policies", nil)
+		d.Set("policies", sg.Policies)
 	}
 	if sg.Rules != nil {
 		d.Set("rules", []map[string]interface{}{{"max_server_per_host": sg.Rules.MaxServerPerHost}})

--- a/openstack/resource_openstack_compute_servergroup_v2.go
+++ b/openstack/resource_openstack_compute_servergroup_v2.go
@@ -94,41 +94,36 @@ func resourceComputeServerGroupV2Create(ctx context.Context, d *schema.ResourceD
 		policy = policies[0]
 	}
 
+	createOpts := ComputeServerGroupV2CreateOpts{
+		servergroups.CreateOpts{
+			Name:   name,
+			Policy: policy,
+		},
+		MapValueSpecs(d),
+	}
+
 	rulesVal, rulesPresent := d.GetOk("rules")
-	var createOpts ComputeServerGroupV2CreateOpts
-	computeClient.Microversion = "2.64"
 	if policy == "anti-affinity" && rulesPresent {
-		maxServerPerHost := expandComputeServerGroupV2RulesMaxServerPerHost(rulesVal.([]interface{}))
-		createOpts = ComputeServerGroupV2CreateOpts{
-			servergroups.CreateOpts{
-				Name:   name,
-				Policy: policy,
-				Rules: &servergroups.Rules{
-					MaxServerPerHost: maxServerPerHost,
-				},
-			},
-			MapValueSpecs(d),
-		}
-	} else {
-		createOpts = ComputeServerGroupV2CreateOpts{
-			servergroups.CreateOpts{
-				Name:   name,
-				Policy: policy,
-			},
-			MapValueSpecs(d),
+		computeClient.Microversion = "2.64"
+		createOpts.CreateOpts.Rules = &servergroups.Rules{
+			MaxServerPerHost: expandComputeServerGroupV2RulesMaxServerPerHost(rulesVal.([]interface{})),
 		}
 	}
 
 	log.Printf("[DEBUG] openstack_compute_servergroup_v2 create options: %#v", createOpts)
 	newSG, err := servergroups.Create(computeClient, createOpts).Extract()
 	if err != nil {
+		// return an error right away
+		if createOpts.CreateOpts.Rules != nil {
+			return diag.Errorf("Error creating openstack_compute_servergroup_v2 %s: %s", name, err)
+		}
+
 		log.Printf("[DEBUG] Falling back to legacy API call due to: %#v", err)
 		// fallback to legacy microversion
-		computeClient.Microversion = ""
 		createOpts = ComputeServerGroupV2CreateOpts{
 			servergroups.CreateOpts{
 				Name:     name,
-				Policies: policies,
+				Policies: expandComputeServerGroupV2Policies(computeClient, rawPolicies),
 			},
 			MapValueSpecs(d),
 		}

